packages/kbn-security-hardening/console.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/* eslint-disable no-console */
+
+// From https://www.ascii-code.com/characters/control-characters,
+// but explicitly allowing the range \u0008-\u000F (line breaks, tabs, etc.)
+const CONTROL_CHAR_REGEXP = new RegExp('[\\u0000-\\u0007\\u0010-\\u001F]', 'g');
+
+export const unsafeConsole = {
+  debug: console.debug.bind(console),
+  error: console.error.bind(console),
+  info: console.info.bind(console),
+  log: console.log.bind(console),
+  trace: console.trace.bind(console),
+  warn: console.warn.bind(console),
+};
+
+function callWithSanitizedArgs(func: Function, ...args: any[]) {
+  const cleanedArgs = args.map(function (arg) {
+    if (typeof arg !== 'string') return arg;
+    return escapeControlChars(arg);
+  });
+  func.apply(console, cleanedArgs);
+}
+
+if (process.env.NODE_ENV === 'production') {
```

Comment:
I was able to manually test this by adding additional console outputs denoting when the console functions were being overridden or not. Do we want to add a single log when the console is overridden (or maybe when it is not overridden)?

Reply:
We could add a line when we patch the console. OTOH, it would be directly to the console (not our logging system) so it won't be visible when looking at logs for deployments not using the default (stdout/console) appender, so I'm not sure.

Reply:
Curious to get your opinion here when you review @azasypkin

Reply:
> Do we want to add a single log when the console is overridden (or maybe when it is not overridden)?

I think it wouldn't hurt to add a log line when we override the native `console`, even if it has limited value :man_shrugging: